assets/js/modules/analytics-4/components/widgets/TopTrafficSourceDrivingPurchasesWidget.js

```diff
@@ -0,0 +1,229 @@
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ trafficSourceReportOptions ] );
+
+		const totalPurchasesReportErrors = select(
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ totalPurchasesReportOptions ] );
+
+		if ( trafficSourceReportErrors && totalPurchasesReportErrors ) {
+			return [ trafficSourceReportErrors, totalPurchasesReportErrors ];
+		}
+
+		return (
+			trafficSourceReportErrors || totalPurchasesReportErrors || undefined
+		);
+	} );
+
+	const loading = useSelect( ( select ) =>
+		hasDetectedEvent
+			? ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ totalPurchasesReportOptions ]
+			  ) ||
+			  ( ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ trafficSourceReportOptions ]
+			  ) &&
+					hasPurchases )
+			: undefined
+	);
+
+	const makeFilter = ( dateRange, dimensionIndex ) => ( row ) =>
+		get( row, `dimensionValues.${ dimensionIndex }.value` ) === dateRange;
+
+	// Prevents running a filter on `report.rows` which could be undefined.
+	const { rows: totalPurchasesReportRows = [] } = totalPurchasesReport || {};
+	const { rows: trafficSourceReportRows = [] } = trafficSourceReport || {};
+
+	const topTrafficSource =
+		trafficSourceReportRows.filter( makeFilter( 'date_range_0', 1 ) )[ 0 ]
+			?.dimensionValues?.[ 0 ].value || '-';
+
+	const currentTotalPurchases =
+		parseInt(
+			totalPurchasesReportRows.filter(
+				makeFilter( 'date_range_0', 0 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const currentTopTrafficSourcePurchases =
+		parseInt(
+			trafficSourceReportRows.filter(
+				makeFilter( 'date_range_0', 1 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+	const relativeCurrentTopTrafficSourcePurchases = currentTotalPurchases
+		? currentTopTrafficSourcePurchases / currentTotalPurchases
+		: 0;
+
+	const previousTotalPurchases =
+		parseInt(
+			totalPurchasesReportRows.filter(
+				makeFilter( 'date_range_1', 0 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const previousTopTrafficSourcePurchases =
+		parseInt(
+			trafficSourceReportRows.filter(
+				makeFilter( 'date_range_1', 1 )
+			)[ 0 ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const relativePreviousTopTrafficSourcePurchases = previousTotalPurchases
+		? previousTopTrafficSourcePurchases / previousTotalPurchases
+		: 0;
+
+	const format = {
+		style: 'percent',
+		signDisplay: 'never',
+		maximumFractionDigits: 1,
+	};
+
+	return (
+		<MetricTileText
+			Widget={ Widget }
+			widgetSlug={ KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_PURCHASES }
+			metricValue={ topTrafficSource }
+			metricValueFormat={ format }
+			subText={
+				// eslint-disable-next-line @wordpress/valid-sprintf
+				sprintf(
+					/* translators: %d: Percentage of leads for the current top traffic source compared to the number of total leads for all traffic sources. */
+					__( '%s of total purchases', 'google-site-kit' ),
```

Comment:
Please, fix the comment for the translation string to have correct placeholders.